desktop-src/direct3dhlsl_current/hlsl-method-getDimensions.md

```diff
@@ -0,0 +1,349 @@
+      out uint width,
+      out uint height,
+      out uint elements );
+```
+
+#### RWTexture3D
+```syntax
+// RWTexture3D
+void RWTexture3D<ElementType>::GetDimensions(
+      out uint width,
+      out uint height,
+      out uint depth );
+```
+
+#### RWBuffer
+```syntax
+// RWBuffer
+void RWBuffer::GetDimensions(
+      out uint dim );
+```
+
+#### RWByteAddressBuffer
+```syntax
+// RWByteAddressBuffer
+void RWByteAddressBuffer::GetDimensions(
+      out uint dim );
+```
+
+#### RWStructuredBuffer
+```syntax
+// RWStructuredBuffer
+void RWStructuredBuffer<ElementType>::GetDimensions(
+      out uint numStructs,
+      out uint stride );
+```
+
+#### AppendStructuredBuffer
+```syntax
+// AppendStructuredBuffer
+void AppendStructuredBuffer<ElementType>::GetDimensions(
+      out uint numStructs,
+      out uint stride );
+```
+
+#### ConsumeStructuredBuffer
+```syntax
+// ConsumeStructuredBuffer
+void ConsumeStructuredBuffer<ElementType>::GetDimensions(
+      out uint numStructs,
+      out uint stride );
+```
+
+## Return Value
+
+None.
+
+## Parameters
+
+| Parameter | Description |
+| - | - |
+| in [`uint mipLevel`](#uint-miplevel) | A zero-based index that identifies the mipmap level. If this argument is not used, the first mip level is assumed. |
+| out [`<UintFloat> width`](#uintfloat-width) | The texture width, in texels. |
+| out [`<UintFloat> numberOfLevels`](#uintfloat-numberOfLevels) | The number of mipmap levels. |
+| out [`<UintFloat> elements`](#uintfloat-elements) | The number of elements in an array. |
+| out [`<UintFloat> height`](#uintfloat-height) | The texture height, in texels. |
+| out [`<UintFloat> depth`](#uintfloat-depth) | The texture depth, in texels. |
+| out [`uint numberOfSamples`](#uint-numberofsamples) | The number of samples. |
+| out [`uint dim`](#uint-dim) | The length, in bytes, of the buffer. |
+| out [`uint numStructs`](#uint-numstructs) | The number of structures in the resource. |
+| out [`uint stride`](#uint-stride) | The stride, in bytes, of each structure element. |
+
+### `uint mipLevel`
+
+A zero-based index that identifies the mipmap level. If this argument is not used, the first mip level is assumed.
+
+### `<UintFloat> width`
+
+The texture width, in texels.
+
+Can be uint or float.
+
+### `<UintFloat> numberOfLevels`
+
+The number of mipmap levels.
+
+Can be uint or float.
+
+### `<UintFloat> elements`
+
+The number of elements in an array.
+
+Can be uint or float.
+
+### `<UintFloat> height`
+
+The texture height, in texels.
+
+Can be uint or float.
+
+### `<UintFloat> depth`
+
+The texture depth, in texels.
+
+Can be uint or float.
+
+### `uint numberOfSamples`
+
+The number of samples.
+
+### `uint dim`
+
+The length, in bytes, of the buffer.
+
+### `uint numStructs`
+
+The number of structures in the resource.
+
+### `uint stride`
+
+The stride, in bytes, of each structure element.
+
+## Remarks
+
+None.
+
+## Example
+
+```HLSL
+// None.
+```
+
+## Supported Shader Models
```

Comment:
This table looks about right for Sample() and some related methods, but not for GetDimensions().  GetDimensions() should be supported at every stage on every shader model.

Reply:
We should remove compatibility currently.